Assets/MixedRealityToolkit/Services/MixedRealityToolkit.cs

```diff
@@ -526,99 +538,68 @@ private void EnsureMixedRealityRequirements()
         /// </summary>
         public static MixedRealityToolkit Instance
         {
             get
             {
-                if (IsInitialized)
+                if (activeInstance != null)
                 {
-                    return instance;
-                }
-
-                if (Application.isPlaying && !searchForInstance)
-                {
-                    return null;
+                    return activeInstance;
                 }
 
-
                 var objects = FindObjectsOfType<MixedRealityToolkit>();
-                searchForInstance = false;
-                MixedRealityToolkit newInstance;
 
                 switch (objects.Length)
                 {
                     case 0:
-#if UNITY_EDITOR
-                        // Generating a new instance during app shutdown in the editor allows instances to persist into edit mode, which may cause unwanted behavior
-                        if (isApplicationQuitting)
-                        {
-                            Debug.LogWarning("Trying to find instance during application shutdown.");
-                            return null;
-                        }
-#endif
-                        Debug.Assert(!newInstanceBeingInitialized, "We shouldn't be initializing another MixedRealityToolkit unless we errored on the previous.");
-                        newInstanceBeingInitialized = true;
-                        newInstance = new GameObject(nameof(MixedRealityToolkit)).AddComponent<MixedRealityToolkit>();
                         break;
+
                     case 1:
-                        newInstanceBeingInitialized = false;
-                        newInstance = objects[0];
+                        RegisterInstance(objects[0]);
                         break;
+
                     default:
-                        newInstanceBeingInitialized = false;
-                        Debug.LogError($"Expected exactly 1 {nameof(MixedRealityToolkit)} but found {objects.Length}.");
-                        return null;
+                        foreach (MixedRealityToolkit toolkit in objects)
+                        {
+                            RegisterInstance(toolkit);
+                        }
+                        break;
                 }
 
-                Debug.Assert(newInstance != null);
-
-                if (!isApplicationQuitting)
-                {
-                    // Setup any additional things the instance needs.
-                    newInstance.InitializeInstance();
-                }
-                else
+                if (activeInstance == null)
                 {
-                    // Don't do any additional setup because the app is quitting.
-                    instance = newInstance;
-                    newInstanceBeingInitialized = false;
+                    Debug.LogError("Didn't find / couldn't create an active instance of mixed reality toolkit. You can add one to your scene via 'Mixed Reality Toolkit->Add to Scene and Configure'");
```

Comment:
This is not an error in my opinion. If we try and use a null instance, we'll get a null reference exception, but if it's null and we null check it, we have enough information there that we don't need to throw an additional error log.